projects/ngx-formentry/src/form-entry/value-adapters/obs-adapter-helper.ts

```diff
@@ -489,20 +489,39 @@ export class ObsAdapterHelper {
   getGroupPayload(node: NodeBase) {
     const nodeAsGroup: GroupNode = node as GroupNode;
 
-    let childrenPayload = [];
+    // Get existing obs
+    let childrenPayload =
```

Comment:
Suggestion: it seems like it would be better to have `childrenPayload` initially as an object keyed by obs UUID rather than a list then instead of all the `find()` calls we can just do `childrenPayload[obs.uuid]`. Then, before constructing the group object, we can just flatten the object into an array of objects.

Reply:
I can do something similar to this, but the reason why this will not work directly is because I am adding other options to this array that may not have a uuid.

Reply:
Ah, ok.